### ISSUE TITLE
Support for using string variables

### ISSUE THREAD

**Issue by Diego-MX** [pydata/numexpr#282]
Thank you for a great tool.  
I'm trying to use a character variable to resemble an `if-else` statement. 
```
a_dict = {"a":3, "b":4, "c":5, "side":"hyp"}
evaluate("(side == 'hyp')*c", a_dict)

 File "/home/diego/Code/anaconda3/lib/python3.6/site-packages/numexpr/necompiler.py", line 686, in getType
    raise ValueError("unknown type %s" % a.dtype.name)

ValueError: unknown type str96
```

Is it possible to do this?  
Thanks

**Comment by robbmcleod** [comment]
Hi,

I think you want to use `where`.  It uses the same conventions as `numpy.where`:

https://docs.scipy.org/doc/numpy/reference/generated/numpy.where.html

Also make sure you only use `bytes` variables with NumExpr. In Python 3 `str` is unicode and that's not supported in NumExpr.   I.e.

    a_dict = {"a":3, "b":4, "c":5, "side":b"hyp"}
    ne.evaluate("(side == b'hyp')*c", local_dict=a_dict)

(see #281 which was just fixed a couple hours ago).

**Comment by Diego-MX** [comment]
Thanks.  That's very useful.  And also the suggestion for `where`.

**Comment by robbmcleod** [comment]
No problem.

Closing as addressed with a patch in 2.6.3 release.
